tests/Aspire.Dashboard.Tests/Model/ResourceViewModelTests.cs

```diff
@@ -1,19 +1,36 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Collections.Immutable;
 using Aspire.Dashboard.Model;
 using Aspire.ResourceService.Proto.V1;
 using Google.Protobuf.WellKnownTypes;
 using Microsoft.Extensions.Logging.Abstractions;
 using Xunit;
-
+using DiagnosticsHealthStatus = Microsoft.Extensions.Diagnostics.HealthChecks.HealthStatus;
 namespace Aspire.Dashboard.Tests.Model;
 
 public sealed class ResourceViewModelTests
 {
     private static readonly DateTime s_dateTime = new(2000, 12, 30, 23, 59, 59, DateTimeKind.Utc);
     private static readonly BrowserTimeProvider s_timeProvider = new(NullLoggerFactory.Instance);
 
+    [Theory]
+    [InlineData(KnownResourceState.Starting, null, null)]
+    [InlineData(KnownResourceState.Starting, null, new string[]{})]
+    [InlineData(KnownResourceState.Starting, null, new string?[]{null})]
+    // we don't have a Running + HealthReports null case because that's not a valid state - by this point, we will have received the list of HealthReports
+    [InlineData(KnownResourceState.Running, DiagnosticsHealthStatus.Healthy, new string[]{})]
+    [InlineData(KnownResourceState.Running, DiagnosticsHealthStatus.Healthy, new string?[] {"Healthy"})]
+    [InlineData(KnownResourceState.Running, DiagnosticsHealthStatus.Unhealthy, new string?[] {null})]
+    [InlineData(KnownResourceState.Running, DiagnosticsHealthStatus.Degraded, new string?[] {"Healthy", "Degraded"})]
```

Comment:
Do we need tests for other states like `Exited`, or `FailedToStart`, `building` etc?

Reply:
We don’t, all states other than Running are treated the same

Reply:
> We don’t, all states other than Running are treated the same

Maybe tests to check for that?

Reply:
That's definitionally true, but I can add these tests. Assuming they just need to go into main?

Reply:
added this in #6461